### PR TITLE
fix(types): backward compatibility fixes

### PR DESCRIPTION
**Pull request by zxbodya**
#### Description

Improving backward compatibility with "old" typescript type definitions, by adding aliases and in some cases adding more types.

### CODE REVIEW

File: src/card/index.ts [hunk 1]
```diff
@@ -17,3 +19,5 @@ export {
   Root as StyledWrapper,
 } from './styled-components';
 export * from './types';
+/** @deprecated To be removed in future versions */
+type hasThumbnail = CardProps['hasThumbnail'];
```
Comment (chasestarr) [inline src/card/index.ts:23]: should CardComponents also be aliased?

File: src/checkbox/index.ts [hunk 2]
```diff
@@ -21,3 +23,5 @@ export { STATE_TYPE, STYLE_TYPE, LABEL_PLACEMENT } from './constants';
 
 // Flow
 export * from './types';
+/** @deprecated use CheckboxReducerState instead. To be removed in future versions.*/
+export type initialState = CheckboxReducerState;
```
Comment (chasestarr) [inline src/checkbox/index.ts:27]:
```suggestion
export type StateT = CheckboxReducerState;
```

File: src/link/index.tsx [hunk 3]
```diff
@@ -47,3 +47,6 @@ export const StyledLink = withWrapper<typeof Link, { animateUnderline?: boolean
       );
     }
 );
+
+/** @deprecated To be removed in future versions.*/
+export type LinkProps = React.ComponentProps<typeof StyledLink>;
```
Comment (chasestarr) [inline src/link/index.tsx:52]: I couldn't find a reference to this, is it required?

File: src/select/index.ts [hunk 4]
```diff
@@ -4,6 +4,9 @@ Copyright (c) Uber Technologies, Inc.
 This source code is licensed under the MIT license found in the
 LICENSE file in the root directory of this source tree.
 */
+import type { Optgroups, Options } from './types';
+import { SelectProps } from '../../legacy-dts/select';
```
Comment (chasestarr) [inline src/select/index.ts]: it looks like this is not used in the file, required?

File: src/timezonepicker/types.ts [hunk 5]
```diff
@@ -28,6 +28,9 @@ export type Timezone = {
    */
   offset: number;
 };
+export type TimezoneOverrides = {
```
Comment (chasestarr) [inline src/timezonepicker/types.ts:31]:
```suggestion
export type TimezonePickerOverrides = {
```

File: src/timezonepicker/types.ts [hunk 6]
```diff
@@ -45,9 +48,7 @@ export type TimezonePickerProps = {
   mapLabels?: (option: Option) => string;
   /** Callback for when the timezone selection changes. */
   onChange?: (value?: Timezone | null) => unknown;
-  overrides?: {
-    Select?: Override;
-  };
+  overrides?: TimezoneOverrides;
```
Comment (chasestarr) [inline src/timezonepicker/types.ts:51]:
```suggestion
  overrides?: TimezonePickerOverrides;
```

File: src/checkbox/index.ts (hunk 2 above)
Comment (zxbodya) [reply src/checkbox/index.ts:27]: this is alias to replace old type definition from here:  https://github.com/uber/baseweb/blob/master/legacy-dts/checkbox/index.d.ts#L21-L24 so initialState is indeed a correct name, but I just noticed it is not actually exported… -- will remove this

File: src/link/index.tsx (hunk 3 above)
Comment (zxbodya) [reply src/link/index.tsx:52]: it is for type that was exported here - https://github.com/uber/baseweb/blob/master/legacy-dts/link/index.d.ts#L4-L11

File: src/timezonepicker/types.ts (hunk 5 above)
Comment (zxbodya) [reply src/timezonepicker/types.ts:31]: that is what we have in old type definitions - https://github.com/uber/baseweb/blob/master/legacy-dts/timezonepicker/index.d.ts#L12-L14

Comment (zxbodya) [reply src/timezonepicker/types.ts:31]: but yeah, looks like a good idea to rename it here and create deprecated alias with old name

**Review by chasestarr — APPROVED** [review]

**Pull request merged**